src/snmp_core.cc

```diff
@@ -445,6 +447,8 @@ snmpConstructReponse(SnmpRequest * rq)
     if (RespPDU != nullptr) {
         snmp_build(&rq->session, RespPDU, rq->outbuf, &rq->outlen);
         comm_udp_sendto(rq->sock, rq->from, rq->outbuf, rq->outlen);
+        // XXX: setup fd_table[] properly so we can use FD_WRITE_METHOD()
+        fd_table[rq->sock].bytesWritten(rq->outlen);
```

Comment:
The new code should not be used when comm_udp_sendto() fails.

```suggestion
        const auto result = comm_udp_sendto(rq->sock, rq->from, rq->outbuf, rq->outlen);
        if (result > 0) {
            // XXX: setup fd_table[] properly so we can use FD_WRITE_METHOD()
            fd_table[rq->sock].bytesWritten(result);
        }
```